src/tslint.ts

```diff
@@ -91,6 +91,10 @@ class Linter {
             sourceFile = getSourceFile(this.fileName, this.source);
         }
 
+        if (sourceFile === undefined) {
+            throw new Error("Invalid source code: " + this.fileName);
```

Comment:
can we be more descriptive here and suggest why the input is considered invalid? how about:

```ts
throw new Error(`Invalid source file "${this.fileName}". Ensure that the files supplied to lint have a .ts or .tsx extension.`);
```